docs/appendices/release-notes/5.8.4.rst

```diff
@@ -68,10 +68,13 @@ Fixes
   array typed column on left hand side of the arguments to return invalid
   results.
 
-- Fixed an issue that caused ``=`` :ref:`ANY <sql_any_array_comparison>`
+- Fixed an issue that caused ``=`` or ``!=`` :ref:`ANY <sql_any_array_comparison>`
   operator to throw a ``ClassCastException`` when the right hand side argument
```

Comment:
LGTM, have a question:

[Fix](https://github.com/crate/crate/pull/16553/commits/0a31f0bebd7b3957161179544816731993a210df) for `=` 
was about fixing all `refMatchesAnyArrayLiteral` overrides in all `AnyOperator` subclasses.

This PR fixes `!=`, ie fixes `literalMatchesAnyArrayRef` override but only for AnyNegOperator. 
1. Shouldn't we fix all `literalMatchesAnyArrayRef` overrides?
2. Shouldn't we mention all fixed operators in the change entry? At least for already merged fix.

Reply:
Thank you @BaurzhanSakhariev, this is the current behaviour on `master`:
```
cr> select * from t where [1] != any(a);
ClassCastException[class java.util.ArrayList cannot be cast to class java.lang.Number (java.util.ArrayList and java.lang.Number are in module java.base of loader 'bootstrap')]
cr> select * from t where [1] < any(a);
UnsupportedFeatureException[Cannot use `< ANY` if left side is an array]
cr> select * from t where [1] like any(a);
UnsupportedFunctionException[Unknown function: ([1] LIKE ANY(doc.t.a)), no overload found for matching argument types: (integer_array, integer_array_array_array). Possible candidates: any_like(text, array(text)):boolean]
```

I am fixing the first one and keeping UnsupportedFunctionExceptions which can be addressed upon request.

> Shouldn't we fix all literalMatchesAnyArrayRef overrides?

The one being fixed is the last one remaining that throws ClassCastExceptions.

> Shouldn't we mention all fixed operators in the change entry? At least for already merged fix.

I thought all are mentioned, could you clarify?

Reply:
> UnsupportedFunctionException

Ah ok, I missed that they are not supported yet.

> all are mentioned

Right, didn't realize that we have a ref covering all. Sorry for the noise.